### PR TITLE
Add check for correct functioning of S3

### PR DESCRIPTION
**Pull request by Sprachprofi**
Tries to connect to the given bucket with the given credentials - this ensures the credentials work and the bucket exists.

Tries to write test.txt to the bucket and to read it back - this ensures our user has read and write access to the bucket. 

Also fixes the default hostname suggested.

### CODE REVIEW

**Comment by ineffyble** [comment]
It seems to me that this will break using S3-compatible non-AWS backends. Can we refactor to address that?

File: lib/tasks/mastodon.rake [hunk 1]
```diff
@@ -177,35 +177,54 @@ namespace :mastodon do
...
-            q.modify :strip
+            env['AWS_SECRET_ACCESS_KEY'] = prompt.ask('S3 secret key:') do |q|
+              q.required true
+              q.modify :strip
+            end
+          
+            begin
+              Aws.config.update({region: env['S3_REGION'], credentials: Aws::Credentials.new(env['AWS_ACCESS_KEY_ID'], env['AWS_SECRET_ACCESS_KEY'])})
+              bucket = Aws::S3::Resource.new.bucket(env['S3_BUCKET'])
+              write_obj = bucket.object("test.txt")
```
Comment (nightpool) [inline lib/tasks/mastodon.rake:216]: I think this should use an auto-generated name to prevent conflicts.

**Comment by nightpool** [comment]
> It seems to me that this will break using S3-compatible non-AWS backends. Can we refactor to address that?

The only issue here is that it's using `S3_REGION` instead of `S3_HOSTNAME`, correct? So that's the only thing to address.

**Comment by ineffyble** [comment]
> The only issue here is that it's using `S3_REGION` instead of `S3_HOSTNAME`, correct? So that's the only thing to address.

That's the big one since it's forcing an AWS hostname, but there's potentially other nuances with particular backends that I'm not across.

**Comment by mjankowski** [comment]
Closing due to no activity in ~1 year -- but this strikes me as potentially useful. If still desired, can you rebase this and reopen?

**Pull request closed without merging**